esda/moran_local_mv.py

```diff
@@ -0,0 +1,375 @@
+            the covariance statistics are not overwhelmed by any single
+            covariate's large variance.
+        mvquads : bool
+            whether to calculate the classification based on the multivariable
+            quadrant classification or the univariate quadrant classification,
+            like a classical Moran's I. When mvquads is True, the variables are labelled as:
+            - label 1: observations with large y - rho * x that also have large Wy values. 
+            - label 2: observations with small y - rho * x values that also have large Wy values.
+            - label 3: observations with small y - rho * x values that also have small Wy values. 
+            - label 4: observations with large y - rho * x values that have small Wy values.
+            When mvquads is false, we ignore the rho * x in the above classification. 
+
+
+        Attributes
+        ----------
+        W      : The weights matrix inputted, but row standardized
+        D      : The "design" matrix used in computation. If X is
+                      not None, this will be [1 y X]
+        R      : the "response" matrix used in computation. Will
+                      always be the same shape as D and contain [1, Wy, Wy, ....]
+        DtDi   : empirical parameter covariance matrix
+                      the P x P matrix describing the variance and covariance
+                      of y and X.
+        P      : the number of parameters. 1 if X is not provided.
+        lmos_   : the N,P matrix of multivariable LISA statistics.
+                      the first column, lmos[:,1] is the LISAs corresponding
+                      to the relationship between Wy and y conditioning on X.
+        rlmos_  : the (N, permutations, P+1) realizations from the conditional
+                  randomization to generate reference distributions for
+                  each Local Moran statistic. rlmos_[:,:,1] pertain to
+                  the reference distribution of y and Wy.
+        quads_  : the (N, P) matrix of quadrant classifications for the
+                  part-regressive relationships. quads[:,0] pertains to
+                  the relationship between y and Wy. The mean is not classified,
+                  since it's just binary above/below mean usually.
+        partials_: the (N,2,P+1) matrix of part-regressive contributions.
+                    The ith slice of partials_[:,:,i] contains the
+                    partial regressive contribution of that covariate, with
+                    the first column indicating the part-regressive outcome
+                    and the second indicating the part-regressive design.
+                    The partial regression matrix starts at zero, so
+                    partials_[:,:,0] corresponds to the partial regression
+                    describing the relationship between y and Wy.
+        """
+        self._mvquads = mvquads
+        y = np.asarray(y).reshape(-1, 1)
+        W.transform = "r"
```

Comment:
Can you ensure support of Graph? This still assumes W only.

Reply:
Yes! I think I've done this with the following check: 

```python
        if hasattr(W, "to_W"):
            W = W.to_W()
```

Reply:
The rest of esda is Graph-native now, so there's no conversion happening. If we can keep it that way, I'd be happier but understand that it may be a larger undertaking. It is up to you.

Reply:
This should now be done. 

One thing of note... if we were to swap `w.sparse * x` for `w.sparse @ x` in `libpysal.weights.spatial_lag.lag_spatial()`, then all instances of `lag_spatial` would work on Graph, too, while preserving backwards compatibility...